src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -2440,6 +2445,88 @@ bool AdaptDispatch::DoConEmuAction(const std::wstring_view /*string*/) noexcept
     return false;
 }
 
+// Method Description:
+// - DECDLD - Downloads one or more characters of a dynamically redefinable
+//   character set (DRCS) with a specified pixel pattern. The pixel array is
+//   transmitted in sixel format via the returned StringHandler function.
+// Arguments:
+// - fontNumber - The buffer number into which the font will be loaded.
+// - startChar - The first character in the set that will be replaced.
+// - eraseControl - Which characters to erase before loading the new data.
+// - cellMatrix - The character cell width (sometimes also height in legacy formats).
+// - fontSet - The screen size for which the font is designed.
+// - fontUsage - Whether it is a text font or a full-cell font.
+// - cellHeight - The character cell height (if not defined by cellMatrix).
+// - charsetSize - Whether the character set is 94 or 96 characters.
+// Return Value:
+// - a function to receive the pixel data or nullptr if parameters are invalid
+ITermDispatch::StringHandler AdaptDispatch::DownloadDRCS(const size_t fontNumber,
+                                                         const VTParameter startChar,
+                                                         const DispatchTypes::DrcsEraseControl eraseControl,
+                                                         const DispatchTypes::DrcsCellMatrix cellMatrix,
+                                                         const DispatchTypes::DrcsFontSet fontSet,
+                                                         const DispatchTypes::DrcsFontUsage fontUsage,
+                                                         const VTParameter cellHeight,
+                                                         const DispatchTypes::DrcsCharsetSize charsetSize)
+{
+    // If we're a conpty, we're just going to ignore the operation for now.
+    // There's no point in trying to pass it through without also being able
+    // to pass through the character set designations.
+    if (_pConApi->IsConsolePty())
+    {
+        return nullptr;
+    }
+
+    // The font buffer is created on demand.
+    if (!_fontBuffer)
+    {
+        _fontBuffer = std::make_unique<FontBuffer>();
+    }
```

Comment:
~Do we need to delay setting this member until the font has been totally constructed? We could make one here, move it into the lambda, and only move it into final storage when it is finalized.~

~Or, is it not a concern that somebody might begin a DRCS, fail, and start a new one? I suppose that we always set the state bits, so... maybe it doesn't matter.~ :smile:

Reply:
~More tightly considered: it is not possible to emit sixel data into a font that did not pass EraseControl/Attributes/StartChar setup, so we will never be appending sixel data into a font that already has some sixel data.~

Reply:
~Is it possible to emit some sixel data, bail out, and then later start a "new" DRCS exchange and inherit the previous sixel data?~

Reply:
You can safely ignore this entire line of questioning! The first thing we do in SetEraseControl is mark the matrix as invalid, and it is legal to add glyphs to an existing font. Teach me to read bottom up eh.